### PR TITLE
[prettier] feat: add name to SupportOption

### PR DESCRIPTION
**Pull request by tyankatsu0105**
@ikatyang @ifiokjr @ffflorian and @sosukesuzuki 
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [ ] Test the change in your own code. (Compile and run.)
- [ ] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <https://prettier.io/docs/en/api.html#prettiergetsupportinfo>

```ts
import prettier from "prettier";
console.log(prettier.getSupportInfo().options.map((item: any) => item.name));
```

```bash
[
  'arrowParens',
  'bracketSpacing',
  'cursorOffset',
  'endOfLine',
  'filepath',
  'htmlWhitespaceSensitivity',
  'insertPragma',
  'jsxBracketSameLine',
  'jsxSingleQuote',
  'parser',
  'pluginSearchDirs',
  'plugins',
  'printWidth',
  'proseWrap',
  'quoteProps',
  'rangeEnd',
  'rangeStart',
  'requirePragma',
  'semi',
  'singleQuote',
  'tabWidth',
  'trailingComma',
  'useTabs',
  'vueIndentScriptAndStyle'
]
```

### CODE REVIEW

File: types/prettier/index.d.ts [hunk 1]
```diff
@@ -324,6 +350,7 @@ export interface SupportOptionDefault {
 }
 
 export interface SupportOption {
+    name: SupportOptionName;
```
Comment (tyankatsu0105) [inline types/prettier/index.d.ts:353]: or
```suggestion
    name: string;
```

**Comment by danger-public** [comment]
Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## prettier ([<kbd>unpkg</kbd>](https://unpkg.com/browse/prettier@latest/))
was missing the following properties:
1. The declaration doesn't match the JavaScript module 'prettier'. Reason:
The JavaScript module can be called or constructed, but the declaration module cannot.

The most common way to resolve this error is to use 'export =' syntax.
To learn more about 'export =' syntax, see https://www.typescriptlang.org/docs/handbook/modules.html#export--and-import--require.
1. re
1. src
1. t
1. tildeTrimReplace

<details>
<summary>as well as these 171 other properties...</summary>
<p>caretTrimReplace, comparatorTrimReplace, apiDescriptor, supportsColor, commonDeprecatedHandler, commonInvalidHandler, levenUnknownHandler, createSchema, Schema, AliasSchema, AnySchema, ArraySchema, BooleanSchema, recordFromArray, mapFromArray, createAutoChecklist, partition, isInt, comparePrimitive, normalizeDefaultResult, normalizeValidateResult, normalizeDeprecatedResult, normalizeTransferResult, normalizeForwardResult, normalizeRedirectResult, ChoiceSchema, NumberSchema, IntegerSchema, StringSchema, defaultDescriptor, defaultUnknownHandler, defaultInvalidHandler, defaultDeprecatedHandler, normalize, Normalizer, matchToToken, ast, code, keyword, shouldHighlight, getChalk, codeFrameColumns, SEMVER_SPEC_VERSION, parse, valid, clean, SemVer, inc, diff, compareIdentifiers, rcompareIdentifiers, major, minor, patch, compare, compareLoose, rcompare, sort, rsort, gt, lt, eq, neq, gte, lte, cmp, Comparator, Range, toComparators, satisfies, maxSatisfying, minSatisfying, minVersion, validRange, ltr, gtr, outside, prerelease, intersects, coerce, minimatch, parseSync, parseString, parseFromFiles, parseFromFilesSync, posix, win32, flatten, splitWhen, isEnoentCodeError, createDirentFromStats, unixify, makeAbsolute, escape, removeLeadingDotSegment, isInteger, find, exceedsLimit, escapeNode, encloseBrace, isInvalidBrace, isOpenOrClose, reduce, isObject, hasRegexChars, isRegexChar, escapeRegex, toPosixSlashes, removeBackslashes, supportsLookbehinds, isWindows, escapeLast, removePrefix, wrapOutput, isStaticPattern, isDynamicPattern, convertToPositivePattern, convertToNegativePattern, isNegativePattern, isPositivePattern, getNegativePatterns, getPositivePatterns, getBaseDirectory, hasGlobStar, endsWithSlashGlobStar, isAffectDepthOfReadingPattern, expandPatternsWithBraceExpansion, expandBraceExpansion, getPatternParts, makeRe, convertPatternsToRe, matchAny, merge, isString, isEmpty, array, errno, fs, path, pattern, stream, string, generate, convertPatternsToTasks, getNegativePatternsAsPositive, groupPatternsByBaseDirectory, convertPatternGroupsToTasks, convertPatternGroupToTask, read, FILE_SYSTEM_ADAPTER, createFileSystemAdapter, Settings, stat, statSync, IS_SUPPORT_READDIR_WITH_FILE_TYPES, readdirWithFileTypes, readdir, scandir, scandirSync, isFatalError, isAppliedFilter, replacePathSegmentSeparator, joinPathSegments, walk, walkSync, walkStream, DEFAULT_FILE_SYSTEM_ADAPTER, extract, strip, parseWithComments, print</p>
</details>

<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 22f7f28b291835c12bcd4285f16639887ae6d01e
</p>

**Comment by tyankatsu0105** [comment]
Oh. 
I don't know mention are automatically send.
Sorry.

File: types/prettier/index.d.ts (hunk 1 above)
Comment (tyankatsu0105) [reply types/prettier/index.d.ts:353]: Which should I do?

Comment (sosukesuzuki) [reply types/prettier/index.d.ts:353]: I think that `string` is better because any strings can be option name for external plugins.

Comment (tyankatsu0105) [reply types/prettier/index.d.ts:353]: I see.

Comment (tyankatsu0105) [reply types/prettier/index.d.ts:353]: done :)

**Review by sosukesuzuki — APPROVED** [review]
LGTM

**Pull request merged**